task/src/main/java/com/flipkart/flux/impl/task/TaskExecutor.java

```diff
@@ -78,7 +78,12 @@ protected Event run() throws Exception {
 		}
 		final Object returnObject = result.getKey();
 		if (returnObject != null) {
-			return new Event(outputeEventName, returnObject.getClass().getCanonicalName(), Event.EventStatus.triggered, stateMachineId, objectMapper.writeValueAsString(returnObject), MANAGED_RUNTIME);
+            Event event = (Event) returnObject;
+            event.setName(outputeEventName);
```

Comment:
I get that you have to use the class loader specific object mapper in local jvm task; but why creating half an event here and half there ?